src/System.Data.SqlClient/tests/ManualTests/SQL/SqlBulkCopyTest/AdjustPrecScaleForBulkCopy.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Data.SqlTypes;
+using Xunit;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public class AdjustPrecScaleForBulkCopy
+    {
+        [CheckConnStrSetupFact]
+        public static void RunTest()
+        {
+            SqlDecimal value = BulkCopySqlDecimalToTable(new SqlDecimal(0), 1, 0, 2, 2);
+            Assert.Equal("0.00", value.ToString());
+
+            value = BulkCopySqlDecimalToTable(new SqlDecimal(10.0), 10, 1, 4, 2);
+            Assert.Equal("10.00", value.ToString());
+
+            value = BulkCopySqlDecimalToTable(new SqlDecimal(10.00), 7, 2, 2, 0);
+            Assert.Equal("10", value.ToString());
+
+            value = BulkCopySqlDecimalToTable(new SqlDecimal(12.345), 5, 3, 3, 1);
+            Assert.Equal("12.3", value.ToString());
+
+            value = BulkCopySqlDecimalToTable(new SqlDecimal(123.45), 10, 2, 4, 1);
+            Assert.Equal("123.4", value.ToString());
+
+            try
+            {
+                BulkCopySqlDecimalToTable(new SqlDecimal(111.00), 7, 2, 2, 0);
```

Comment:
You can use Assert.Throws here. There's also the DataTestUtility.AssertThrowsWrapper, which has more options.